### PR TITLE
NewWithBracesFixer - fix by adding BraceClassInstantiationTransformer

### PR DESCRIPTION
**Pull request by sstok**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | no
| Fixed tickets | #1000
| License       | MIT

The NewWithBracesFixerTest fails because of this change but I can't figure out why.

### CODE REVIEW

File: Symfony/CS/Tests/Tokenizer/Transformer/BraceClassInstantiationTest.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * @author Sebastiaans Stok <s.stok@rollerscapes.net>
+ */
+class BraceClassInstantiationTest extends AbstractTransformerTestBase
+{
+    /**
+     * @dataProvider provideProcessCases
+     */
+    public function testProcess($source, array $expectedTokens)
+    {
+        $tokens = Tokens::fromCode($source);
```
Comment (keradus) [inline Symfony/CS/Tests/Tokenizer/Transformer/BraceClassInstantiationTest.php]: why not `$this->makeTest($source, $expectedTokens)` ?

File: Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+            if (!$tokens[$tokens->getNextMeaningfulToken($index)]->equals(array(T_NEW))) {
+                continue;
+            }
+
+            $openIndex = $index;
+            $closeIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $openIndex);
+
+            if ($tokens[$tokens->getNextMeaningfulToken($closeIndex)]->equalsAny(array(array(T_OBJECT_OPERATOR), array(T_DOUBLE_COLON)))) {
+                $tokens[$openIndex]->override(array(CT_BRACE_CLASS_INSTANTIATION_OPEN, '(', $tokens[$openIndex]->getLine()));
```
Comment (keradus) [inline Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php]: there is no line at master branch

File: Symfony/CS/Tests/Tokenizer/Transformer/BraceClassInstantiationTest.php (hunk 1 above)
Comment (sstok) [inline Symfony/CS/Tests/Tokenizer/Transformer/BraceClassInstantiationTest.php]: I just copy-pasted from another test :)

File: Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php (hunk 2 above)
Comment (sstok) [inline Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php]: Copy paste ;)

File: Symfony/CS/Tests/Tokenizer/Transformer/BraceClassInstantiationTest.php (hunk 1 above)
Comment (keradus) [inline Symfony/CS/Tests/Tokenizer/Transformer/BraceClassInstantiationTest.php]: copy-paste is wrong, isn't it?
You take provided data style from one test but test method from other.
please, use makeTest

File: Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php (hunk 2 above)
Comment (keradus) [inline Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php]: > Copy paste ;)

Copy paste :(:(:(

no, `Token->override()` does not set the line.

Comment (keradus) [inline Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php]: BTW, Travis blow out on this...

Comment (sstok) [inline Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php]: https://github.com/fabpot/PHP-CS-Fixer/blob/master/Symfony/CS/Tokenizer/Token.php#L372

Comment (keradus) [inline Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php]: Try using FriendsOfPHP's repo instead of Fabien's private fork ;)

Comment (sstok) [inline Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php]: I'm pretty sure that will work better yes :')

**Comment by sstok** [comment]
OK, should be good now.

**Comment by sstok** [comment]
3 times a charm :)

**Comment by sstok** [comment]
Travis failure is caused by hhvm, and unrelated to this PR.

**Comment by keradus** [comment]
looks great !
:+1:

**Comment by keradus** [comment]
> Travis failure is caused by hhvm, and unrelated to this PR.

Yeah, the reason is here: #1004

File: Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * Transform braced class instantiation braces in `(new Foo())` into CT_BRACE_CLASS_INSTANTIATION_OPEN
+ * and CT_BRACE_CLASS_INSTANTIATION_CLOSE.
+ *
+ * @author Sebastiaans Stok <s.stok@rollerscapes.net>
+ *
+ * @internal
+ */
+class BraceClassInstantiation extends AbstractTransformer
```
Comment (keradus) [inline Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php]: Please add missing `Transformer` suffix

Comment (sstok) [inline Symfony/CS/Tokenizer/Transformer/BraceClassInstantiation.php]: Done

**Comment by keradus** [comment]
Thank you @sstok.

**Pull request closed without merging**